### ISSUE TITLE
During incremental copy, the last row is always updated

### ISSUE THREAD

**Issue by timmysuh** [singer-io/tap-postgres#104]
I cannot think of a reason why we have ">=" instead of just ">" for INCREMENTAL replication logic?

It seems that we always end up updating the last row. Why is this a desired behavior?

if replication_key_value:
                    select_sql = """SELECT {}
                                    FROM {}
                                    WHERE {} >= '{}'::{}
                                    ORDER BY {} ASC""".format(','.join(escaped_columns),
                                                              post_db.fully_qualified_table_name(schema_name, stream['table_name']),
                                                              post_db.prepare_columns_sql(replication_key), replication_key_value, replication_key_sql_datatype,
                                                              post_db.prepare_columns_sql(replication_key))

**Comment by INGCRENGIFO** [comment]
I have the same doubt.

**Comment by tmonks** [comment]
I agree, it should be ">". I've tested this change and am no longer getting the duplicates. Hoping we can get this change made to master.

**Comment by NicolasRisi** [comment]
In fact, it's normal to have ">=" to avoid loss of data.
If 2 row are written sequentially but have the same "updated_at" value, and the extraction is made between the 2 writes, you will lose data.

**Comment by tmonks** [comment]
What's the best way to avoid duplicated rows then? I haven't run into this issue with any of the other taps I've worked with.

**Comment by NicolasRisi** [comment]
The target must manage that and upsert or merge the New data.

**Comment by bback99** [comment]
@NicolasRisi Any recommends when original table doesn't have the PK but target has PK?

so WHERE {} >= '{}'::{}

this makes duplicated key when rerun the query in this case.
